### PR TITLE
Setup database migrations, initialize db and implement models

### PR DESCRIPTION
**Pull request by bunsenmcdubbs**
- [x] Update README with new PostgreSQL dependency and config steps
- [x] Add support for database migrations (run as cli flag on server binary)
- [x] Initialize database (with first migration) to set up schema and tables
- [x] Update README with instructions on how to run database migrations
- [x] Update README with instructions on how to create database migrations
- [x] Add models for created tables
- [x] Add tests for models
- [Future TODO] Automatically create an admin user from environment variables if none is present

We can now run database migrations using the `--migrate` flag! Example: `go run main.go --migrate` or `go build . && ./SponsorshipPortal --migrate` (You have to have the correct database config/connection strings as environment variables).

Open questions:
- [x] How does this get integrated/automated with Docker/k8s/deployment? cc @illegalprime @ehsanmasdar 
  - [x] **Migrations:** Application containers will check for and apply any pending migrations before boot up at each deploy. Applications will also report unhealthy on `/_health` endpoint while database is in a dirty (incomplete) migration state. (bbd2a0c)
  - [x] **Database creation:** ??? Ideally we can create a new database (`CREATE DATABASE`) for each PR automatically so we can isolate issues.
✅ Ideally HackGT/biodomes#16 is implemented and takes care of this.
  - [Future TODO] **Seeding:** ??? It would be nice to have a dev script that automatically populated the database (both on local dev environments and any deployed non-production instances) with some test data.

Continues/will finish move to PostgreSQL (#44)

### CODE REVIEW

**Comment by illegalprime** [comment]
How does this work when we're migrating from tiedot to postgres?

File: migrations/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# MIgrations
```
Comment (illegalprime) [inline migrations/README.md]: typo?

Comment (bunsenmcdubbs) [reply migrations/README.md]: yes

**Comment by bunsenmcdubbs** [comment]
@illegalprime This is essentially a regression (technically #55 was the regression). tiedot (and any persistence) has been completely removed and we are re-implementing that functionality with postgres here

**Comment by bunsenmcdubbs** [comment]
Waiting for HackGT/biodomes#16 to be merged and deployed before `wants: postgres` stanza is supported.

**Pull request merged**